File: lingting-polaris-grpc/src/main/java/live/lingting/polaris/grpc/resolver/ResolverContext.java

package live.lingting.polaris.grpc.resolver;

import com.tencent.polaris.api.pojo.ServiceKey;
import com.tencent.polaris.client.api.SDKContext;

import java.net.URI;

public class ResolverContext {

	private URI targetUri;

	private SDKContext context;

	private ServiceKey sourceService;

	public URI getTargetUri() {
		return targetUri;
	}

	public SDKContext getContext() {
		return context;
	}

	public ServiceKey getSourceService() {
		return sourceService;
	}

	public static ResolverContextBuilder builder() {
		return new ResolverContextBuilder();
	}

	public static final class ResolverContextBuilder {

		private URI targetUri;

		private SDKContext context;

		private ServiceKey sourceService;

		private ResolverContextBuilder() {
		}

		public ResolverContextBuilder targetUri(URI targetUri) {
			this.targetUri = targetUri;
			return this;
		}

		public ResolverContextBuilder context(SDKContext context) {
			this.context = context;
			return this;
		}

		public ResolverContextBuilder sourceService(ServiceKey sourceService) {
			this.sourceService = sourceService;
			return this;
		}

		public ResolverContext build() {
			ResolverContext resolverContext = new ResolverContext();
			resolverContext.context = this.context;
			resolverContext.targetUri = this.targetUri;
			resolverContext.sourceService = this.sourceService;
			return resolverContext;
		}

	}

}

File: lingting-polaris-grpc/src/main/kotlin/live/lingting/polaris/grpc/resolver/ResolverContext.kt
package live.lingting.polaris.grpc.resolver

import com.tencent.polaris.api.pojo.ServiceKey
import com.tencent.polaris.client.api.SDKContext
import java.net.URI

class ResolverContext {
    var targetUri: URI? = null
        private set

    var context: SDKContext? = null
        private set

    var sourceService: ServiceKey? = null
        private set

    class ResolverContextBuilder {
        private var targetUri: URI? = null

        private var context: SDKContext? = null

        private var sourceService: ServiceKey? = null

        fun targetUri(targetUri: URI?): ResolverContextBuilder {
            this.targetUri = targetUri
            return this
        }

        fun context(context: SDKContext?): ResolverContextBuilder {
            this.context = context
            return this
        }

        fun sourceService(sourceService: ServiceKey?): ResolverContextBuilder {
            this.sourceService = sourceService
            return this
        }

        fun build(): ResolverContext {
            val resolverContext = ResolverContext()
            resolverContext.context = this.context
            resolverContext.targetUri = this.targetUri
            resolverContext.sourceService = this.sourceService
            return resolverContext
        }
    }

    companion object {
        fun builder(): ResolverContextBuilder {
            return ResolverContextBuilder()
        }
    }
}
